agent/cache-types/streaming.go

```diff
@@ -89,6 +91,21 @@ func (s *Subscriber) run() {
 			return
 		}
 
+		// Check to see if this is a special "reload stream" message before processing
+		// the event.
+		if event.GetReloadStream() {
+			fmt.Printf("[WARN] got ACL reset event\n")
+			s.handler.Reset()
+
+			// Cancel the context to end the streaming call and create a new one.
+			s.lock.Lock()
+			s.cancelFunc()
+			s.ctx, s.cancelFunc = context.WithCancel(context.Background())
+			s.lock.Unlock()
+
+			goto START
```

Comment:
Jumping back to start with no backoff makes me wonder if there are error cases where this can get into a hot loop.

I don't _think_ it should in normal use but I wonder if we should have some kind of exponential backoff on this just incase we somehow get into a loop because of a server bug or something?

Reply:
I think without some sort of event buffer to keep the acl update events around in the publisher we couldn't get stuck in a hot loop here since that's the only source of the reload event - that said, I agree it's probably better to add a backoff here now to avoid something unexpected.